labs-ios-starter/Model Controller/UserController+API.swift

```diff
@@ -12,41 +12,48 @@ import OktaAuth
 
 extension UserController {
 
-    func fetchTopics(completion: @escaping (TopicResults?) -> Void) {
-        guard let oktaCredentials = getOktaAuth() else { return }
-
+    func fetchTopics(isMock: Bool = false, completion: @escaping (TopicResults?) -> Void) {
         let requestURL = baseURL.appendingPathComponent("topics").appendingPathComponent("topics")
         var request = URLRequest(url: requestURL)
-        request.addValue("Bearer \(oktaCredentials.accessToken)", forHTTPHeaderField: "Authorization")
 
-        URLSession.shared.dataTask(with: request) { data, response, error in
+        if !isMock {
+            guard let oktaCredentials = getOktaAuth() else { return }
+            request.addValue("Bearer \(oktaCredentials.accessToken)", forHTTPHeaderField: "Authorization")
+        }
+
+        dataLoader.loadData(using: request) { data, response, error in
             if let error = error {
-                NSLog("Error fetching topics: \(error)")
-                completion(nil)
-                return
-            }
+                    NSLog("Error fetching topics: \(error)")
+                    completion(nil)
+                    return
+                }
 
-            guard let data = data else {
-                NSLog("No topic data for topics request")
-                completion(nil)
-                return
-            }
+                guard let data = data else {
+                    NSLog("No topic data for topics request")
+                    completion(nil)
+                    return
+                }
 
-            do {
-                let topics = try JSONDecoder().decode(TopicResults.self, from: data)
-                DispatchQueue.main.async { completion(topics) }
-            } catch {
-                NSLog("Error decoding topics data: \(error)")
-                completion(nil)
+                do {
+                    let topics = try JSONDecoder().decode(TopicResults.self, from: data)
+                    DispatchQueue.main.async { completion(topics) }
+                } catch {
+                    NSLog("Error decoding topics data: \(error)")
+                    completion(nil)
+                }
```

Comment:
This might need to get formatted a little bit because it looks like there is extra tab here.